java/java.editor.base/src/org/netbeans/modules/java/editor/base/semantic/SemanticHighlighterBase.java

```diff
@@ -871,7 +872,18 @@ public Void visitClass(ClassTree tree, Void p) {
             }
             scan(tree.getExtendsClause(), null);
             scan(tree.getImplementsClause(), null);
-
+            try {
+                while (!TokenUtilities.textEquals(tl.currentToken().text(), "{")) {// NOI18N
```

Comment:
My suggestion here would be to:
1. read the Permitted subclasses
2. if non-null/non-empty, do simply firstIdentifierToken("permits"), and add the token to contextKeywords
3. run scan(<permitted>, null)
(so that the classes inside the permits clause are highlighted).